src/userpreferences/repository.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Handle fetching and saving user preferences
+ * @alpha
+ */
+export interface UserPreferencesRepository {
+    /**
+     * Get user preferences
+     *
+     * @param key - name of the key containing the data
+     * @returns the data
+     */
+    get<T = any>(key?: string): T;
```

Comment:
Leave this be until the discussion with Johan about generic type has come to some conclusion, but: _if_ we remove the generic type from set, it should also be removed from here:

```suggestion
    get(key?: string): string;
```

Reply:
Yes let us conclude the discussion with Johan first

Reply:
Leave this as it is, since we keep the generic type 👍